File: gym-backend/src/main/java/com/bequre/pojo/Equipment.java
package com.bequre.pojo;


import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.time.LocalDate;

@Data
@Schema(description = "设备实体")
public class Equipment {

    @NotNull
    @Schema(name = "id", description = "ID")
    @TableId(value = "id", type = IdType.AUTO)
    private Integer id;

    @NotNull
    @Schema(name = "name", description = "名称")
    private String name;

    @NotNull
    @Schema(name = "quantity", description = "数量")
    private Integer quantity;

    @NotNull
    @Schema(name = "status", description = "状态")
    private String status;

    @NotNull
    @Schema(name = "updateDate", description = "更新日期")
    private LocalDate updateDate;

    @NotNull
    @Schema(name = "notes", description = "备注")
    private String notes;


}

File: gym-front/src/types/equipment.ts
/**
 * Equipment，设备实体
 */
export interface Equipment {
    /**
     * ID
     * 添加的请求不需要这个参数
     */
    id?: number;
    /**
     * 名称
     */
    name: string;
    /**
     * 数量   改成string好处理一些
     */
    quantity: string;
    /**
     * 状态
     */
    status: string;
    /**
     * 更新日期
     */
    updateDate?: Date;
    /**
     * 备注
     */
    notes: string;
    //[property: string]: any;
}

/**
 * 器材分页条件查询实体
 */
export interface EquipmentQuery {

    /**
     * 页码
     */
    pageNo: number;
    /**
     * 每页条数
     */
    pageSize: number;
    /**
     * 器材名称
     */
    name?: string | null;
    /**
     * 器材状态
     */
    status?: string | null;

}